src/components/new-safe/create/steps/StatusStep/useSafeCreation.ts

```diff
@@ -61,34 +64,65 @@ export const useSafeCreation = (
     setIsCreating(true)
     dispatch(closeByGroupKey({ groupKey: SAFE_CREATION_ERROR_KEY }))
 
-    try {
-      const tx = await getSafeCreationTxInfo(provider, pendingSafe, chain, wallet)
-
-      const safeParams = getSafeDeployProps(
-        {
-          threshold: pendingSafe.threshold,
-          owners: pendingSafe.owners.map((owner) => owner.address),
-          saltNonce: pendingSafe.saltNonce,
-        },
-        (txHash) => createSafeCallback(txHash, tx),
-        chain.chainId,
-      )
-
-      await createNewSafe(provider, safeParams)
-      setStatus(SafeCreationStatus.SUCCESS)
-    } catch (err) {
-      const _err = err as EthersError
-      const status = handleSafeCreationError(_err)
-
-      setStatus(status)
-
-      if (status !== SafeCreationStatus.SUCCESS) {
-        dispatch(showSafeCreationError(_err))
+    const { owners, threshold, saltNonce } = pendingSafe
+    const ownersAddresses = owners.map((owner) => owner.address)
+
+    if (willRelay) {
+      try {
+        const taskId = await createNewSafeViaRelayer(chain, ownersAddresses, threshold, saltNonce)
```

Comment:
Is it possible to move the relayer logic into a separate function and doing the check inside the `useEffect` since they don't share most of the dependencies (relaying doesn't need a wallet or a provider) e.g.

```
const createSafeViaRelayer = () => {}
const createSafe = () => {}

useEffect(() => {
    if (willRelay) {
        void createSafeViaRelayer()
    } else {
        void createSafe()
    }
}
```